lib/internal/http2/compat.js

```diff
@@ -513,12 +514,9 @@ class Http2ServerResponse extends Stream {
     if (stream !== undefined &&
         stream.destroyed === false &&
         stream.headersSent === false) {
-      options = options || Object.create(null);
-      const state = this[kState];
       const headers = this[kHeaders];
-      headers[HTTP2_HEADER_STATUS] = state.statusCode;
-      if (stream.finished === true)
-        options.endStream = true;
```

Comment:
I could not find any reason that this would be needed. All our tests pass without it and if it is truly needed then there should be a test to cover it. Would appreciate any input!

Reply:
The stream.finished check here is an optimization to avoid sending an unnecessary empty DATA frame.

Reply:
@jasnell Right but what's the situation where stream.finished could be true in this particular code? We don't have any test cases that generate a situation like this (as you can see in the coverage since this code path is never hit).

Reply:
Hmm... is `stream.finished` even a thing? As far as I can tell, it's just always `undefined`. There's `this.finished` or `stream._writableState.finished`. Can they even write in either of those cases?